syntax = "proto3";

package queue;

service QueueService {
    rpc AddtoQueue (AddtoQueueRequest) returns (AddtoQueueResponse);
    rpc RequestFromQueue (GetFromQueueRequest) returns (GetFromQueueResponse);
}

message AddtoQueueRequest {
    string orderId = 1;
    string name = 2;
    string number = 3;
    string expirationDate = 4;
    string cvv = 5;
    string contact = 6;
}

message AddtoQueueResponse {
    bool response = 1;
}

message GetFromQueueRequest {
    string orderId = 1;
}

message GetFromQueueResponse {
    string orderId = 1;
    string name = 2;
    string number = 3;
    string expirationDate = 4;
    string cvv = 5;
    string contact = 6;
}